sage/pallet/src/lib.rs

```diff
@@ -395,14 +392,14 @@ pub mod pallet {
 			to: AccountIdOf<T>,
 			asset_id: AssetIdOf<T, I>,
 		) -> DispatchResult {
+			let from = ensure_signed(origin)?;
+
 			let GeneralConfig { transfer, .. } = GeneralConfigStore::<T, I>::get();
-			let from = match Self::ensure_organizer(origin.clone()) {
-				Ok(organizer) => organizer,
-				_ => {
-					ensure!(transfer.open, Error::<T, I>::TransferClosed);
-					ensure_signed(origin)?
-				},
-			};
+			ensure!(
+				Self::ensure_organizer(&from).is_ok() || transfer.open,
+				Error::<T, I>::TransferClosed
+			);
```

Comment:
Here, we simplify the code a lot and make it more efficient with the new `ensure_organizer`.